actors/builtin/miner/miner_actor.go

```diff
@@ -1869,12 +1866,6 @@ func getVerifyInfo(rt Runtime, params *SealVerifyStuff) *abi.SealVerifyInfo {
 		rt.Abortf(exitcode.ErrForbidden, "too early to prove sector")
 	}
 
-	// Check randomness.
-	challengeEarliest := sealChallengeEarliest(rt.CurrEpoch(), params.RegisteredSealProof)
-	if params.SealRandEpoch < challengeEarliest {
-		rt.Abortf(exitcode.ErrIllegalArgument, "seal epoch %v too old, expected >= %v", params.SealRandEpoch, challengeEarliest)
-	}
-
```

Comment:
Why are we not checking this anymore?

Reply:
This check is incorrect.  This function is only called during prove commit.  The intention behind the check is to ensure the precommit seal rand epoch is not too far back from the precommit epoch.  However by using current epoch it is checking that the difference between prove commit time and precommit seal rand epoch is less than max duration between precommit and precommit seal rand epoch.  So this is overly punitive.

On top of the incorrectness the `params.SealRandEpoch` value comes from an onchain value already correctly checked [here](https://github.com/filecoin-project/specs-actors/pull/878/files#diff-e47fb4c42d791672c9550edb1dc5f969R404), so it is fine to remove this check.